lib/livebook/hubs/team_client.ex

```diff
@@ -244,9 +244,19 @@ defmodule Livebook.Hubs.TeamClient do
     }
   end
 
-  defp build_deployment_group(state, %{id: id, name: name, mode: mode, secrets: secrets}) do
-    secrets = Enum.map(secrets, &build_secret(state, &1))
-    %DeploymentGroup{id: id, name: name, mode: mode, hub_id: state.hub.id, secrets: secrets}
+  defp build_deployment_group(state, deployment_group) do
+    secrets = Enum.map(deployment_group.secrets, &build_secret(state, &1))
+
+    %DeploymentGroup{
+      id: deployment_group.id,
+      name: deployment_group.name,
+      mode: deployment_group.mode,
+      hub_id: state.hub.id,
+      secrets: secrets,
+      clustering: deployment_group.clustering,
+      zta_provider: String.to_atom(deployment_group.zta_provider),
```

Comment:
@cristineguadelupe can you please do a quick test, change this to:

```suggestion
      clustering: "whatever",
      zta_provider: "whatever",
```

And then try to deploy a notebook with this deployment group. Does it work as expected? That's because it can happen that someone is using an old Livebook version and the server sends a clustering or zta_provider option that was defined in a newer version.

Reply:
I have reviewed the code and it all looks like it should, but I thought we should double check.

Reply:
Yes, that works. But we're not using this data for app deployment yet. Just the secrets at the moment.